api/src/legal-documents/domain/usecases/get-legal-document-status-by-user-id.usecase.js

```diff
@@ -18,23 +16,19 @@ const getLegalDocumentStatusByUserId = async ({
   userId,
   service,
   type,
-  userRepository,
   legalDocumentRepository,
   userAcceptanceRepository,
-  featureToggles = config.featureToggles,
+  logger,
 }) => {
   LegalDocumentService.assert(service);
   LegalDocumentType.assert(type);
 
-  const { isLegalDocumentsVersioningEnabled } = featureToggles;
-  if (!isLegalDocumentsVersioningEnabled) {
-    const user = await userRepository.findPixOrgaCgusByUserId(userId);
-    return LegalDocumentStatus.buildForLegacyPixOrgaCgu(user);
-  }
-
   const lastLegalDocument = await legalDocumentRepository.findLastVersionByTypeAndService({ service, type });
 
-  if (!lastLegalDocument) throw new LegalDocumentVersionNotFoundError();
+  if (!lastLegalDocument) {
+    logger.warn(`Unknown legal document version found for ${service} and ${type}`);
+    return LegalDocumentStatus.unknown();
+  }
```

Comment:
Je pense que nous devrions utiliser ici tout simplement la classe `NotFoundError`  https://github.com/1024pix/pix/blob/dev/api/src/shared/domain/errors.js#L472 car avoir une classe `LegalDocumentVersionNotFoundError` dédiée n'apporte rien car on ne veut pas un traitement spécifique dans ce cas puisque que ce cas ne doit pas arriver.

```suggestion
  if (!lastLegalDocument) {
    throw new NotFoundError();
  }
```

Reply:
Nous avons décidé d'enlever l'état unknown, en faveur d'un état "requested" par défaut dans le cas où la version du document n'existe pas (accompagné d'une log en warning)